scapy/contrib/bgp.py

```diff
@@ -136,14 +136,13 @@ def addfield(self, pkt, s, val):
         return s + self.i2m(pkt, val)
 
     def getfield(self, pkt, s):
-        length = self.mask2iplen(struct.unpack(">B", s[0])[0]) + 1
+        length = self.mask2iplen(orb(s[0])) + 1
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
+        mask = orb(m[0])
         mask2iplen_res = self.mask2iplen(mask)
-        ip = "".join(
-            [m[i + 1] if i < mask2iplen_res else b"\x00" for i in range(4)])
+        ip = b"".join(m[i + 1] if i < mask2iplen_res else b"\x00" for i in range(4))
```

Comment:
Will this work with Python 3? I think you need `m[i + 1:i + 2]`, or did I miss something?

Reply:
my bad